controllers/gitrepository_controller.go

```diff
@@ -122,13 +122,13 @@ func (r *GitRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			conditions.WithConditions(
 				sourcev1.IncludeUnavailableCondition,
 				sourcev1.SourceVerifiedCondition,
-				sourcev1.CheckoutFailedCondition,
+				sourcev1.FetchFailedCondition,
 				sourcev1.ArtifactOutdatedCondition,
 				sourcev1.ArtifactUnavailableCondition,
 			),
 			conditions.WithNegativePolarityConditions(
 				sourcev1.ArtifactUnavailableCondition,
-				sourcev1.CheckoutFailedCondition,
+				sourcev1.FetchFailedCondition,
```

Comment:
i don't see where the `negativePolarityConditionsTypes` are being checked. When I check the `summary` helper function I only see `conditionsTypes`check :https://github.com/fluxcd/pkg/blob/f5d963e27b8b4a0058b614f255e985773d4da7da/runtime/conditions/getter.go#L155

Reply:
The information is used during transformation to the target type, see: https://github.com/fluxcd/pkg/blob/11abd078e26f7eaf8c4c415fa1897e1d41993592/runtime/conditions/merge.go#L58